### ISSUE TITLE
After updating to 2.0 from 1.2.2 : git fetch exited with code 128

### ISSUE THREAD

**Issue by remyzorg** [ocaml/opam#3211]
I'm not sure if the problem comes from OPAM or from the package owner, but it used to work well in 1.2.2.  After updating from 1.2.2 to 2.0, I received this error during  `opam update` part __Synchronising development packages__.

```
[ERROR] Could not synchronize /home/remy/.opam/4.05.0/.opam-switch/sources/js_of_ocaml-ppx_deriving_json from
        "git+ssh://git@github.com//ocsigen/js_of_ocaml.git":
        "/usr/bin/git fetch -q ssh://git@github.com//ocsigen/js_of_ocaml.git" exited with code 128
```

Then I ran the git command from .opam-switch/sources/js_of_ocaml-ppx_deriving_json

```
/usr/bin/git fetch -q ssh://git@github.com//ocsigen/js_of_ocaml.git
fatal: remote error: 
  /ocsigen/js_of_ocaml is not a valid repository name
  Email support@github.com for help
```

The problem seems to come from the repo name that has two slashes instead of one : 
`//ocsigen/js_of_ocaml.git`. The same situation appears with different pinned packages that have, or have not, their own git repo.

Here is one of the incriminated OPAM file. 
https://github.com/ocsigen/js_of_ocaml/blob/master/js_of_ocaml-ppx_deriving_json.opam

And also, just in case, here is my config :

```
# opam config report
# opam-version      2.0.0~beta5 
# self-upgrade      no
# system            arch=x86_64 os=linux os-distribution=fedora os-version=27
# solver            builtin-mccs+glpk
# install-criteria  -removed,-count[version-lag,request],-count[version-lag,changed],-changed
# upgrade-criteria  -removed,-count[version-lag,solution],-new
# jobs              4
# repositories      1 (http) (default repo at 2018-02-06 10:04)
# pinned            26 (git)
# current-switch    4.05.0
```

**Comment by AltGr** [comment]
Different, but might have some relationship with #3204
opam 1.2 allowed various formats for pinning targets, e.g. "git@github.com:/ocsigen/js_of_ocaml.git"; while opam 2.0 normalises  all of them (internally: the above is still accepted from the command-line). One of the reasons was that the 1.2 required the kind of pinning to be stored separately.

So there might be a rewriting issue during the normalisation that led to the double slash and the error.

**Comment by AltGr** [comment]
The rewrite should be fixed by  #3212. In your case, it's too late, though, so you'll have to fix the pinning targets of your packages manually to remove the extra `/` — sorry about that.

Thanks for reporting!

**Comment by remyzorg** [comment]
Thank you for the quick answer and fix 👍
